cibuildwheel/linux.py

```diff
@@ -130,6 +130,26 @@ def build(options: BuildOptions) -> None:
 
             call(['docker', 'start', container_name])
 
+            if options.before_all:
+                call(
+                    ['docker', 'exec', '-i', container_name] + shell_cmd,
+                    universal_newlines=True,
+                    input='''
+                        PS4='    + '
+
+                        set -o errexit
+                        set -o xtrace
+
+                        {environment_exports}
```

Comment:
Do we want to run `CIBW_ENVIRONMENT` multiple times?

So far, we were only doing this once, so doing it multiple times seems unexpected. So if getting the result from some command is expensive, this will be done twice?
We might have users depending on this, or not?

For example, I have `CCACHE_BASEDIR=$(python -c "import tempfile; import os; print(os.path.realpath(tempfile.gettempdir())))` in `CIBW_ENVIRONMENT`. This is not so expensive in my case, but I don't know what's out there for other users.

Reply:
> Do we want to run `CIBW_ENVIRONMENT` multiple times?
> 
> So far, we were only doing this once, so doing it multiple times seems unexpected. So if getting the result from some command is expensive, this will be done twice?
> We might have users depending on this, or not?

Are you shure? For me enviroment is executed at least one per python version

Reply:
Ah, yes, my mistake. That's already since the previous refactoring, then.
Maybe we should warn people in the docs, though?